cpp/src/arrow/util/tracing_internal.cc

```diff
@@ -18,6 +18,10 @@
 #include "arrow/util/tracing_internal.h"
 #include "arrow/util/tracing.h"
 
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/sysinfo.h>
```

Comment:
Two things: 1) can we use the C++ includes where available (e.g. `<cstdio>` instead of `<stdio.h`>)? 2) can we keep the system includes alphabetically ordered?

Reply:
Why were these includes added to this particular file anyways?  Is this a case where we weren't following IWYU and you are making it more clear?